lib/lexer.js

```diff
@@ -142,15 +144,20 @@ Lexer.prototype._output = function() {
 Lexer.prototype._embed = function() {
   var tag = this._next() + this._next()
     , ch = this._peek();
+
   if(tag == '<' + this._embedChar) {
     this._inEmbed = true;
     if(ch == '=' || ch == '-') return this._token(Token.EMBED, this._next());
     else return this.nextToken();
   } else if(tag == this._embedChar + '>') {
     this._inEmbed = false;
     return this.nextToken();
+  } else if(tag + ch == "-" + this._embedChar + '>') {
+    this._next();
```

Comment:
Sorry for not explaining very well. It's actually not working. If you add a non-whitespace character after the embed token (which should be fine), the parser fails with an error. Here's an example of a valid template that fails (notice I added "s are great" after the embed line from your example) :

```
<% posts.forEach(function (c) { %>
    - <%= c -%>s are great
<% }) %>
```

Here's the failure backtrace:

```
/Users/evan/Code/ajs/lib/parser.js:82
  throw new Error(msg + " at line " + (line || this.lexer._line) + ", column " + (col || this.lexer._col));
  ^

Error: Unexpected token: T_NAME (s) at line 2, column 45
    at Parser._error (/Users/evan/Code/ajs/lib/parser.js:82:9)
    at Parser._tokenError (/Users/evan/Code/ajs/lib/parser.js:86:8)
    at Parser._unexpected (/Users/evan/Code/ajs/lib/parser.js:90:8)
    at Parser._semicolon (/Users/evan/Code/ajs/lib/parser.js:110:78)
    at Parser._prog1 (/Users/evan/Code/ajs/lib/parser.js:623:18)
    at Parser._simpleStatement (/Users/evan/Code/ajs/lib/parser.js:231:42)
    at Parser._statement (/Users/evan/Code/ajs/lib/parser.js:141:22)
    at Parser._embed (/Users/evan/Code/ajs/lib/parser.js:213:42)
    at Parser._statement (/Users/evan/Code/ajs/lib/parser.js:130:19)
    at Parser._block (/Users/evan/Code/ajs/lib/parser.js:320:17)
```